pypeit/core/telluric.py

```diff
@@ -2131,52 +2394,22 @@ def show_fit_qa(self, iord):
-                - meta_table: Table containing the meta information for
-                  the telluric model fits
-                - out_table: Table containing the telluric model fits
-                  and the object model fits.
         """
 
-        # Allocate the meta parameter table, ext=1
-        meta_table = table.Table(meta={'name': 'Parameter Values'})
-        meta_table['TOL'] = [self.tol]
-        meta_table['POPSIZE'] = [self.popsize]
-        meta_table['RECOMBINATION'] = [self.recombination]
-        meta_table['TELGRIDFILE'] = [os.path.basename(self.telgridfile)]
+        self.model = self.empty_model_table(self.norders, self.nspec_in,
+                                            n_obj_par=self.max_ntheta_obj)
         if 'output_meta_keys' in self.obj_params:
             for key in self.obj_params['output_meta_keys']:
-                meta_table[key.upper()] = [self.obj_params[key]]
+                if key.lower() in self.datamodel.keys():
+                    setattr(self, key.lower(), self.obj_params[key])
+                elif key.upper() in self.model.keys():
+                    self.model[key.upper()] = self.obj_params[key]
         if self.ech_orders is not None:
-            meta_table['ECH_ORDERS'] = [self.ech_orders]
-
-        # Allocate the output table, ext=2
-        out_table = table.Table(meta={'name': 'Object Model and Telluric Correction'})
-        out_table['WAVE'] = np.zeros((self.norders, self.nspec_in))
-        out_table['TELLURIC'] = np.zeros((self.norders, self.nspec_in))
-        out_table['OBJ_MODEL'] = np.zeros((self.norders, self.nspec_in))
-        out_table['TELL_THETA'] = np.zeros((self.norders, 7))
-        out_table['TELL_PRESS'] = np.zeros(self.norders)
-        out_table['TELL_TEMP'] = np.zeros(self.norders)
-        out_table['TELL_H2O'] = np.zeros(self.norders)
-        out_table['TELL_AIRMASS'] = np.zeros(self.norders)
-        out_table['TELL_RESLN'] = np.zeros(self.norders)
-        out_table['TELL_SHIFT'] = np.zeros(self.norders)
-        out_table['TELL_STRETCH'] = np.zeros(self.norders)
-        out_table['OBJ_THETA'] = np.zeros((self.norders, self.max_ntheta_obj))
-        out_table['CHI2'] = np.zeros(self.norders)
-        out_table['SUCCESS'] = np.zeros(self.norders, dtype=bool)
-        out_table['NITER'] = np.zeros(self.norders, dtype=int)
-        out_table['IND_LOWER'] = self.ind_lower
-        out_table['IND_UPPER'] = self.ind_upper
-        out_table['WAVE_MIN'] = self.wave_grid[self.ind_lower]
-        out_table['WAVE_MAX'] = self.wave_grid[self.ind_upper]
-
-
-        return meta_table, out_table
-
+            self.model['ECH_ORDERS'] = self.ech_orders
+        self.model['IND_LOWER'] = self.ind_lower
+        self.model['IND_UPPER'] = self.ind_upper
+        self.model['WAVE_MIN'] = self.wave_grid[self.ind_lower]
+        self.model['WAVE_MAX'] = self.wave_grid[self.ind_upper]
 
     def assign_output(self, iord):
```

Comment:
This method seems a bit odd, e.g. why aren't this simply assigned when they are first accessible?  But, I'm guessing there is a good reason..

Reply:
This is inherited from the previous version.